### DISCUSSION TITLE
Search by username and banning spammers

### DISCUSSION THREAD

**Discussion by Peepo1** [tosdr/edit.tosdr.org#983] [Ideas]
I've seen spammers use the same account to post spam all across the site. When they do this and you recognize them as a spammer, it's very hard to find every comment where they were advertising or posting links. If there was a username search function, it would make flagging their actions infinitely easier and you can be sure to clean all of them up. 

Also, once you clean up all of the spammer's mess there should be a way to ban their account so they don't keep doing it. 
However, this would require a change in our own Terms of Service to give us that ability. It can say that "spam or unsolicited messages is not allowed on ToS;DR and is considered a breach of the terms. We will email a warning to a user that breaks the terms and if the user breaks the terms a second time, their account will be terminated. If your account is terminated by us, you are not allowed to re-register under a new account." 

This covers three points that currently exist on ToS:DR's own site (1 positive and 2 neutral).
- This service prohibits users sending chain letters, junk mail, spam or any unsolicited messages (neutral)
- User accounts can be terminated after having been in breach of the terms of service repeatedly (positive)
- Users who have been permanently banned from this service are not allowed to re-register under a new account (neutral)

**Comment by DrKJeff16** [comment]
Indeed. The influx in the edit tool is quite annoying.

`Nice article! If you want to learn more about online dating[...]`

**Comment by madoleary** [comment]
Yeah, ugh, spam has been a real challenge for us.

A lot of spam seems to happen in the "comments". We implemented a feature to mark comments as "spam", and then later they're destroyed in a task that runs automatically on a regular basis.

We could implement something that tracks the user's spam count. For example, after three counts of spam and a warning, all of the user's content across the website is destroyed, and the user's account is deleted and/or banned. 

In the longer term, it could be interesting to integrate something like Akismet. This looks cool: https://github.com/jonahb/akismet
